server/sonar-server/src/main/java/org/sonar/server/plugins/ws/InstalledPluginsWsAction.java

```diff
@@ -0,0 +1,110 @@
+import org.sonar.api.utils.text.JsonWriter;
+
+import java.util.Comparator;
+import java.util.List;
+
+import static com.google.common.base.Objects.firstNonNull;
+import static com.google.common.collect.Iterables.filter;
+import static com.google.common.collect.Lists.newArrayList;
+import static java.util.Collections.sort;
+
+/**
+ * Implementation of the {@code installed} action for the Plugins WebService.
+ */
+public class InstalledPluginsWsAction implements PluginsWsAction {
+  private static final String DEFAULT_VERSION = "-";
+  private static final String PROPERTY_KEY = "key";
+  private static final String PROPERTY_NAME = "name";
+  private static final String PROPERTY_VERSION = "version";
+
+  private final PluginRepository pluginRepository;
+
+  public InstalledPluginsWsAction(PluginRepository pluginRepository) {
+    this.pluginRepository = pluginRepository;
+  }
+
+  @Override
+  public void define(WebService.NewController controller) {
+    controller.createAction("installed")
+      .setDescription("Get the list of all the plugins installed on the SonarQube instance, sorted by name")
+      .setSince("5.2")
+      .setHandler(this)
+      .setResponseExample(Resources.getResource(this.getClass(), "example-installed_plugins.json"));
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    List<PluginMetadata> pluginMetadatas = retrieveAndSortPluginMetadata();
+    writeMetadataList(response, pluginMetadatas);
+  }
+
+  private List<PluginMetadata> retrieveAndSortPluginMetadata() {
+    List<PluginMetadata> pluginMetadatas = newArrayList(filter(pluginRepository.getMetadata(), NotCorePluginsPredicate.INSTANCE));
+    sort(pluginMetadatas, PluginMetadataNameComparator.INSTANCE);
+    return pluginMetadatas;
+  }
+
+  private void writeMetadataList(Response response, List<PluginMetadata> pluginMetadatas) {
+    JsonWriter jsonWriter = response.newJsonWriter();
+    jsonWriter.beginArray();
+    for (PluginMetadata pluginMetadata : pluginMetadatas) {
+      writeMetadata(jsonWriter, pluginMetadata);
+    }
+    jsonWriter.endArray();
+    jsonWriter.close();
+  }
+
+  private void writeMetadata(JsonWriter jsonWriter, PluginMetadata pluginMetadata) {
+    jsonWriter.beginObject();
+    jsonWriter.prop(PROPERTY_KEY, pluginMetadata.getKey());
+    jsonWriter.prop(PROPERTY_NAME, pluginMetadata.getName());
+    jsonWriter.prop(PROPERTY_VERSION, firstNonNull(pluginMetadata.getVersion(), DEFAULT_VERSION));
+    jsonWriter.endObject();
+  }
+
+  private enum NotCorePluginsPredicate implements Predicate<PluginMetadata> {
+    INSTANCE;
+
+    @Override
+    public boolean apply(PluginMetadata input) {
+      return !input.isCore();
+    }
+  }
+
+  private enum PluginMetadataNameComparator implements Comparator<PluginMetadata> {
+    INSTANCE;
+
+    @Override
+    public int compare(PluginMetadata o1, PluginMetadata o2) {
+      return o1.getName().compareTo(o2.getName());
```

Comment:
the sort is now part of the WS contract, it should therefor be enforced by the WS implementation (and this order is not part of the method `getMetaData`'s contract so it is weak to rely on it)